src/DynamoCoreWpf/Views/PackageManager/Controls/PackageManagerPackagesControl.xaml.cs

```diff
@@ -149,13 +149,12 @@ private void CloseToastButton_OnClick(object sender, RoutedEventArgs e)
             return;
         }
 
-       
 
         private void DropDownInstallButton_OnClick(object sender, RoutedEventArgs e)
         {
             var button = sender as Button;
-            if (button == null) { return; }
-                       
+            if (button == null || button.DataContext == null) { return; }
```

Comment:
does anything ever test the install to specific folder functionality?

Reply:
It will eventually get to https://github.com/DynamoDS/Dynamo/blob/d8db34eae83e9aaf82326c77644f99b4a1c86bf6/src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs#L749 and in between it is tested if the provided path (from the dialog) is null.

Reply:
Sorry, I meant is there an automated test that exercises that code path?

Reply:
There are 3 tests for the `ExecutePackageDownload`, which both the `install` and `install from folder` point to. The difference between the two is that the `install` will install inside the default package location, whereas the `install from folder` will take user input. But both use the same API and pass a string installPath to the ExecutePackageDownload. 

The 3 tests that are inside the `PackageManagerUITests` are:
- PackageManagerDownloadError
- PackageManagerDownloadsBeforeInstalling
- PackageManagerWarnWhenInstallingPackageTargetingOtherHost